### DISCUSSION TITLE
`add(['.'])` adds the `.git` folder

### DISCUSSION THREAD

**Discussion by agirault** [gitpython-developers/GitPython#1446] [Q&A] [answered]
Hoping to add all files in the directory using `git add .`, including files starting with a period like `.gitignore`.

Calling the following:
```py
repo.index.add(['.'])
```

Stages the `.git` folder and all its history.

**Comment by Byron** [comment, marked as answer]
`repo.index` can't be used for cases like this as it's not a complete implementation of 'pathspecs'. It's a much narrower implementation that can be considered 'expert mode'.

By default, I recommend using `repo.git.add('.')` instead to have the work done by `git` instead.
